src/api/app/jobs/cleanup_events.rb

```diff
@@ -1,5 +1,5 @@
 class CleanupEvents < ActiveJob::Base
   def perform
-    Event::Base.where(project_logged: true, queued: true, undone_jobs: 0).delete_all
+    Event::Base.where(project_logged: true, queued: true, undone_jobs: 0).lock(true).delete_all
```

Comment:
I don't think you're supposed to use pessimistic locking without wrapping this into a transaction...